### PR TITLE
fix linter + strong type + refactor node

### PR DESCRIPTION
**Pull request by didOstap**
fix linter
types
simple functions => array functions
simple promise & callbacks => async/await

### CODE REVIEW

File: server/src/controllers/forgot.password.controllers/update.password.ts [hunk 1]
```diff
@@ -1,28 +1,21 @@
...
 
-export const updatePassword = async (req: Request,
-                                     res: Response,
-                                     next: NextFunction) => {
+export const updatePassword = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
     try {
-        const {token} = req.params;
-        const {password} = req.body;
+        const tokenFromEmail = req.params.token;
+        const password = req.body.password;
```
Comment (InnaKutsik) [inline server/src/controllers/forgot.password.controllers/update.password.ts]: you can use es6 destructuring
```
const {password} = req.body;
```

File: server/src/controllers/post.controllers/create.post.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import {NextFunction, Request, Response} from 'express';
 
-export const create = (req: Request, res: Response, next: NextFunction) => {
-
+export const create = (req: Request, res: Response, next: NextFunction): any => {
+    return 1;
```
Comment (InnaKutsik) [inline server/src/controllers/post.controllers/create.post.ts]: why 1?

File: server/src/controllers/confirm.user.controllers/resend.token.controller.ts [hunk 3]
```diff
@@ -1,53 +1,24 @@
...
+import {resendTokenMessage} from '../../helpers/send.email.message';
 
-export const resend = async (req: Request,
-                             res: Response,
-                             next: NextFunction) => {
+export const resend = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
     try {
-        interface IEmail {
-            email: string;
+        const email: string = req.body.email;
```
Comment (InnaKutsik) [inline server/src/controllers/confirm.user.controllers/resend.token.controller.ts]: you can use es6 destructuring
```
const {email} = req.body;
```

File: server/src/controllers/forgot.password.controllers/check.email.ts [hunk 4]
```diff
@@ -1,79 +1,25 @@
...
+import {Request, Response, NextFunction} from 'express';
+import {sendEmail} from '../../helpers/send.email';
+import {userExist} from '../../common.db.request/user.exist';
+import {forgotPasswordMessage} from '../../helpers/send.email.message';
 
-const existEmail = async (bodyEmail: string, next: NextFunction) => {
+export const checkEmail = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
     try {
-        if (!bodyEmail) {
+        const email: string = req.body.email;
```
Comment (InnaKutsik) [inline server/src/controllers/forgot.password.controllers/check.email.ts]: you can use es6 destructuring

const {email} = req.body;

File: server/src/controllers/confirm.user.controllers/confirm.token.controller.ts [hunk 5]
```diff
@@ -1,20 +1,17 @@
...
 
-export const confirm = async (req: Request,
-                              res: Response,
-                              next: NextFunction) => {
+export const confirm = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
     try {
-        const tokenFromEmail: string = req.params.token;
-
-        const token = await Token.findOne({token: tokenFromEmail});
+        const tokenFromEmail = req.params.token;
```
Comment (InnaKutsik) [inline server/src/controllers/confirm.user.controllers/confirm.token.controller.ts]: you can use es6 destructuring

const { token: tokenFromEmail } = req.params;

File: server/src/controllers/post.controllers/find.all.posts.ts [hunk 6]
```diff
@@ -1,5 +1,5 @@
 import {NextFunction, Request, Response} from 'express';
 
-export const findAll = (req: Request, res: Response, next: NextFunction) => {
-
+export const findAll = (req: Request, res: Response, next: NextFunction): any => {
+    return 1;
```
Comment (InnaKutsik) [inline server/src/controllers/post.controllers/find.all.posts.ts]: why 1?

File: server/src/helpers/hash.password.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+import bcrypt from 'bcrypt';
+
+export const hashPassword = (password: string): string => {
+    const salt: string = bcrypt.genSaltSync(12);
```
Comment (InnaKutsik) [inline server/src/helpers/hash.password.ts]: pls 12 move to constant with clear name

File: server/src/helpers/send.email.ts [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        const transporter = nodemailer.createTransport({
+            service: 'gmail',
+            auth: {
+                user: process.env.EMAIL,
+                pass: process.env.EMAIL_PASS,
+            },
+        });
+
+        const newToken = token.token;
+        const url = `http://localhost:8080/confirm/${newToken}`;
```
Comment (InnaKutsik) [inline server/src/helpers/send.email.ts]: pls use process.env variable instead of hardcode http://localhost:8080/

File: server/tslint.json [hunk 9]
```diff
@@ -1,18 +1,51 @@
...
+  ],
+  "jsRules": {},
+  "rules": {
+    "object-literal-sort-keys": false,
+    "ordered-imports": false,
+    "max-line-length": [
+      true,
+      {
+        "limit": 120,
+        "ignore-pattern": "^import |^export {(.*?)}|class [a-zA-Z]+ implements |"
```
Comment (InnaKutsik) [inline server/tslint.json]: so import can be huge?

File: server/src/controllers/forgot.password.controllers/reset.password.ts [hunk 10]
```diff
@@ -1,11 +1,9 @@
...
-import {Token} from '../../models/token.model';
+import {ITokenModel, Token} from '../../models/token.model';
 
-export const resetPassword = async (req: Request,
-                                    res: Response,
-                                    next: NextFunction) => {
+export const resetPassword = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
     try {
-        const tokenFromEmail: string = req.params.token;
+        const {token: tokenFromEmail}: ITokenModel = req.params.token;
```
Comment (InnaKutsik) [inline server/src/controllers/forgot.password.controllers/reset.password.ts]: const {token: tokenFromEmail}: ITokenModel = req.params.token;

у обєкта token є пропертя token?

File: server/src/helpers/hash.password.ts [hunk 11]
```diff
@@ -0,0 +1,7 @@
+import bcrypt from 'bcrypt';
+
+export const hashPassword = (password: string): string => {
+    const rounds = +`${process.env.ROUNDS}`;
```
Comment (InnaKutsik) [inline server/src/helpers/hash.password.ts]: move to constant, write after import 
const ROUNDS = 12

if you need nubber, you can use Number(process.env.ROUNDS)

and pls give more understandable name

File: server/src/controllers/forgot.password.controllers/reset.password.ts (hunk 10 above)
Comment (didOstap) [reply server/src/controllers/forgot.password.controllers/reset.password.ts]: yes

const TokenSchema: Schema = new Schema({
    user: {type: Schema.Types.ObjectId, ref: 'User', required: true},
    token: {type: String, required: true},
    createdAt: {type: Date, default: Date.now, expires: 43200, required: true},
});

**Pull request merged**